### PR TITLE
New visit dimension  "total interactions" and new action dimension "Interaction position"  and corresponding segments

### PR DESCRIPTION
**Pull request by tsteur**
Adding a small column to track at which position within the visit an interaction was performed. I had first added an action segment "interaction_position" so users could eg only have a look at first 3 pageviews/searches but then I noticed we need to define an `action type` for a segment and because this segment belongs to many types (eg pageurl and sitesearch) I cannot add it to a specific one.

I have tested it manually but couldn't really figure out how to add a test for this. Later I want to have the types configurable so people can define what they understand under "interaction" and eg include events.

ping @mattab

### CODE REVIEW

File: core/DataAccess/LogQueryBuilder/JoinGenerator.php [hunk 1]
```diff
@@ -228,6 +228,12 @@ public function sortTablesForJoin($tA, $tB)
         );
 
         if (is_array($tA) && is_array($tB)) {
+            if (isset($tB['tableAlias']) && isset($tA['joinOn']) && strpos($tA['joinOn'], $tB['tableAlias']) !== false) {
```
Comment (tsteur) [inline core/DataAccess/LogQueryBuilder/JoinGenerator.php]: fix Piwik does not join correctly if 2 custom joins require each other see test

**Comment by mattab** [comment]
not needed for this PR but a note:

* Whenever we display the API token in the UI, would be good to only show the first few characters and on click show the full token. this helps to let other people view the screen (from above shoulder or via screen sharing) but not leak any critical security credential

**Comment by tsteur** [comment]
Yes that's always good. On the marketplace we show the license key only after a click as well.
Maybe you meant to put this comment for the password PR?

**Comment by mattab** [comment]
@tsteur PR looks good! The easiest way to test the feature may be to add the columns in the output of the Live APIs so they will appear in our existing system tests?

**Comment by tsteur** [comment]
done we only need to update custom dimensions plugin before merging to fix test there and resolve conflict

**Pull request merged**